dist/s_all

```diff
@@ -103,6 +103,7 @@ run "python test_tag.py"
 run "./s_mentions" "--warning-only"
 if [ $(command -v evergreen) ] && [ -f ~/.evergreen.yml ]; then
     run "evergreen validate -p wiredtiger ../test/evergreen.yml"
+    run "evergreen validate -p wiredtiger ../test/evergreen_develop.yml"
```

Comment:
Yeah I'll be updating that in the future but I'm happy with what this PR is doing for now.

Reply:
> Can you check this by intentionally seeding a fault in evergreen_develop.yml and running s_all?
> We have the following lines in s_all that check evergreen.yml file is correct and we might need to duplicate for evergreen_develop.yml:

Interestingly, when I run s_all locally without injecting a fault in `evergreen_develop.yml`, I got the below error
```
~/workspace/wiredtiger-changes/dist git:(wt-10256-evg-cfg-for-develop)
$ ./s_all -A -E
dist/s_all run started...
Updating files that include the package version
####################### MESSAGE ############################
s_all run of: "evergreen validate -p wiredtiger ../test/evergreen_develop.yml" resulted in:
    ERROR: load project error(s): retrieving file 'test/evergreen.yml': reading project config: open test/evergreen.yml: no such file or directory
    ../test/evergreen_develop.yml is an invalid configuration
#######################
dist/s_all run finished
```
The `s-all` task in the [PR patch build](https://spruce.mongodb.com/task/wiredtiger_ubuntu2004_s_all_patch_06415c814ac6fd7a6c5a54f1fb560451e83a5efe_639800b20ae606603ef9f2de_22_12_13_04_33_55/logs?execution=0) did not manage to capture the same issue.

The error above seems to indicate it attempted to retrieve the configuration file from the wiredtiger Evergreen project (i.e. `test/evergreen.yml`) and failed to find the file likely because the current directory is `dist`.

Reply:
When injecting a fault in evergreen_develop.yml, the fault was captured in the below error:
```
./s_all -A -E
dist/s_all run started...
Updating files that include the package version
####################### MESSAGE ############################
s_all run of: "evergreen validate -p wiredtiger ../test/evergreen_develop.yml" resulted in:
    ERROR: load project error(s): error unmarshalling yaml strict: yaml: unmarshal errors:
      line 10: field isplay_name not found in type model.copyType
    ../test/evergreen_develop.yml is an invalid configuration
#######################
dist/s_all run finished
```

Reply:
@ajmorton I pushed [8e20231](https://github.com/wiredtiger/wiredtiger/pull/8587/commits/8e2023157abeb4b356280f317a14822c1fd73732) and ran a few tests (with and without injected fault in the Evergreen config file). Please lmk if it looks okay to you. Cheers

Reply:
@ajmorton I broke down the checks into 2 lines as suggested, to make both dash and bash happy. Also fixed a typo in the comment. See [bfe8cf9](https://github.com/wiredtiger/wiredtiger/pull/8587/commits/bfe8cf9e65bb058e993442daa6cad9a5c2536c80)

Reply:
Look perfect, thanks!